### ISSUE TITLE
going from profile to frontal

### ISSUE THREAD

**Issue by nyck33** [iacopomasi/face_specific_augm#31]
I used `py2to3` to get a python 3 version and testing on some profile face images but it seems that when the image starts in profile, it does not rotate it towards frontal.  I guess a GAN frontalization is what I need here.  Or will Face-Pose-Net do this?

**Comment by iacopomasi** [comment]
The frontalization can tolerate medium variations in yaw (maybe 20-30 degree at most); I think there is a logic in the code to avoid frontalizing extremely profile faces otherwise you may get strong artifacts.  Probably the best way to go to frontalize profile faces would be a mixture of CG+GAN so that CG (computer graphics) does the major work and the GAN will make the face realistic and repair the artifacts. Keep in mind if you use GAN for synthesis, the synthesis will be biased to the training data you used for GAN training: although the synthetic face may look realistic you may lose identity traits of the individual. The literature of face GAN synthesis is huge so I advise to go and read that.

**Comment by nyck33** [comment]
@iacopomasi 
Thanks for the reply.  I wanted to use this to augment datasets for gender detection such as this cvlib implementation:  https://github.com/arunponnusamy/gender-detection-keras

However, I have been reading that males and females have different facial characteristics so can you tell me which of the ten basic 3D face models for rendering used in your code are based on males and which are females?  I believe there are ten of them.  Also, if they are not gender specific, is there any way to subsitute other models, ie. if I can find male face 3D models and female face 3D models and use those.  

Visual inspection of some of the models (I think number 3) with a sloped forehead definitely look more male where as some of the others I think tend to feminize faces.  

I actually tested on a small test set (1 male, 1 female) of all the rendered faces at every pose and it was 100 percent correct for male/female detection.
